MatrixSDK/ClientInformation/MXClientInformationService.swift

```diff
@@ -0,0 +1,90 @@
+// 
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+@objcMembers
+public class MXClientInformationService: NSObject {
+
+    private weak var session: MXSession?
+
+    public init(withSession session: MXSession) {
+        self.session = session
+    }
+
+    public func refresh() {
+        guard MXSDKOptions.sharedInstance().enableNewClientInformationFeature else {
+            return removeDataIfNeeded()
+        }
+
+        guard let session = session else {
+            return
+        }
+
+        guard let bundleDisplayName = Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") ?? Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as? String,
+              let appVersion = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String else {
+            return
+        }
+
+        let updatedInfo: [AnyHashable: String] = [
```

Comment:
A small suggestion. It would be nice to break down `refresh` method to make it more readable because it contains multiple `guard` statements. For example, the creation of `updatedInfo` could be extracted into a separate method `createAccountData`

Reply:
Yep good idea.